gst/mqtt/mqttsrc.c

```diff
@@ -445,6 +445,20 @@ gst_mqtt_src_change_state (GstElement * element, GstStateChange transition)
 
   switch (transition) {
     case GST_STATE_CHANGE_PLAYING_TO_PAUSED:
+      g_mutex_lock (&self->mqtt_src_mutex);
+      if (self->is_subscribed) {
+        int sub_ret;
+
+        sub_ret = MQTTAsync_unsubscribe (self->mqtt_client_handle,
+            self->mqtt_topic, &self->mqtt_respn_opts);
+        if (sub_ret != MQTTASYNC_SUCCESS) {
+          GST_ERROR_OBJECT (self,
+              "MQTTAsync_unsubscribe failed with return code %d", sub_ret);
+        }
+
+        self->is_subscribed = FALSE;
```

Comment:
Even if `MQTTAsync_unsubscribe` is failed?

Reply:
> Even if `MQTTAsync_unsubscribe` is failed?

At this point, there is nothing we can do.

Reply:
true..